File: CSharp/DependencyBreakingKatas/B/MarketingCampaign.cs
using System;

namespace Org.Codecop.Dependencies.B
{
    public class MarketingCampaign
    {

        public bool IsActive()
        {
            return MilliSeconds() % 2 == 0;
        }

        private long MilliSeconds()
        {
            return (long) DateTime.Now.TimeOfDay.TotalMilliseconds;
        }

        public bool IsCrazySalesDay()
        {
            return DayOfTheWeek().Equals(DayOfWeek.Friday); ;
        }

        private DayOfWeek DayOfTheWeek()
        {
            return DateTime.Now.DayOfWeek;
        }
    }
}

File: TypeScript/app/b/marketing-campaign.ts
export class MarketingCampaign {

    public isActive(): boolean {
        return this.milliseconds() % 2 == 0;
    }

    public isCrazySalesDay(): boolean {
        return this.dayOfWeek() == 5; // Friday
    }

    private milliseconds(): number {
        const now: Date = new Date();
        return now.getMilliseconds();
    }

    private dayOfWeek(): number {
        const date: Date = new Date();
        return date.getDay();
    }
}
